x/mint/keeper/genesis.go

```diff
@@ -8,11 +8,24 @@ import (
 
 // InitGenesis new mint genesis
 func (keeper Keeper) InitGenesis(ctx context.Context, ak types.AccountKeeper, data *types.GenesisState) {
-	if err := keeper.Minter.Set(ctx, data.Minter); err != nil {
+	if err := keeper.Params.Set(ctx, data.Params); err != nil {
 		panic(err)
 	}
 
-	if err := keeper.Params.Set(ctx, data.Params); err != nil {
+	err := keeper.PreviousRewardEmissionPerUnitStakedToken.Set(
+		ctx,
+		data.PreviousRewardEmissionPerUnitStakedToken,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	err = keeper.PreviousBlockEmission.Set(ctx, data.PreviousBlockEmission)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
also why are all these errors passed around like variables? can't we just panic as soon as there's an error instead of passing it around and checking error states everywhere? isn't it riskier to rely on remembering to check it? also it makes the code very cluttered

Reply:
another go convention. You pass errs back to the caller, so that deep yielding doesn't happen. no `try catch`